src/coreclr/jit/lsrabuild.cpp

```diff
@@ -3841,13 +3851,15 @@ int LinearScan::BuildReturn(GenTree* tree)
                         {
                             hasMismatchedRegTypes = true;
                             regMaskTP dstRegMask  = genRegMask(retTypeDesc.GetABIReturnReg(i));
-                            if (varTypeUsesFloatReg(dstType))
+
+                            if (varTypeUsesIntReg(dstType))
                             {
-                                buildInternalFloatRegisterDefForNode(tree, dstRegMask);
+                                buildInternalIntRegisterDefForNode(tree, dstRegMask);
                             }
                             else
                             {
-                                buildInternalIntRegisterDefForNode(tree, dstRegMask);
+                                assert(varTypeUsesFloatReg(dstType));
```

Comment:
Not necessarily.

I simply added asserts to the places where `kmask` wasn't being handled. This works today since `kmask` is only introduced in lowering and is restricted to exactly one register and is only used as a return value for an intrinsic.

Once we finish expanding the support to take `kmask` as inputs, to support all 8 registers, and to have them appear from user code in importation, then some of these asserts may trigger and support will need to be added at that time.